crates/bevy_animation/src/lib.rs

```diff
@@ -565,8 +565,17 @@ fn run_animation_player(
     }
 }
 
-/// Update `weights` based on weights in `keyframes` at index `key_index`
-/// with a linear interpolation on `key_lerp`.
+/// Update `weights` based on weights in `keyframe` with a linear interpolation
+/// on `key_lerp`.
+fn lerp_morph_weights(weights: &mut [f32], key_lerp: f32, keyframe: impl Iterator<Item = f32>) {
+    let zipped = weights.iter_mut().zip(keyframe);
+    for (morph_weight, keyframe) in zipped {
+        let minus_lerp = 1.0 - key_lerp;
+        *morph_weight = (*morph_weight * minus_lerp) + (keyframe * key_lerp);
```

Comment:
@CGMossa recommended this

```suggestion
        *morph_weight += (keyframe - *morph_weight) * key_lerp;
```